### PR TITLE
feat/Add histograms for duration metrics

### LINKED ISSUE #2351: Add histograms for request metrics
## Is your feature request related to a problem? Please describe.
I'm always frustrated when I'm looking at counters for requests, histograms are much more useful and don't add a lot of cost.

## Describe the solution you'd like
Use `prometheus.Histogram` instead of counters for request metrics.

## Describe alternatives you've considered
Leaving things as is. Meh.

## Additional context
#2344. nspcc-dev/neo-go#2784.

### PR DESCRIPTION
**Pull request by carpawell**
Deprecate `prometheus.Counter` analogs. Closes #2351.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -38,6 +38,11 @@ on timeout, try increasing the value, for example, twice. Also note that the
 execution of commands with the `--await` flag and without an explicitly
 specified time period is now limited to 1 minute. This value can be changed with
 `--timeout` flag.
+Histogram (not counter) RPC/engine operation handling time metrics were added. For
```
Comment (roman-khimov) [inline CHANGELOG.md:41]: That's `migration`, but something should also be written into the `added` section.

**Review by roman-khimov — COMMENTED** [review]
Seems to be OK otherwise.

File: pkg/metrics/engine.go [hunk 2]
```diff
@@ -8,103 +8,197 @@ import (
...
+			Subsystem: engineSubsystem,
+			Name:      "delete_time",
+			Help:      "Engine delete operations handling time",
+		})
+
+		existsDuration = prometheus.NewHistogram(prometheus.HistogramOpts{
+			Namespace: storageNodeNameSpace,
+			Subsystem: engineSubsystem,
+			Name:      "exists_time",
+			Help:      "Engine exists operations handling time",
```
Comment (roman-khimov) [inline pkg/metrics/engine.go:69]: `Engine "exists" operation...`?

Comment (carpawell) [reply pkg/metrics/engine.go:69]: do you mean quoting operation names?

File: pkg/metrics/engine.go [hunk 3]
```diff
@@ -8,103 +8,197 @@ import (
...
-		getDuration                   prometheus.Counter
-		headDuration                  prometheus.Counter
-		inhumeDuration                prometheus.Counter
-		putDuration                   prometheus.Counter
-		rangeDuration                 prometheus.Counter
-		searchDuration                prometheus.Counter
-		listObjectsDuration           prometheus.Counter
-		containerSize                 prometheus.GaugeVec
-		payloadSize                   prometheus.GaugeVec
+		listContainersDuration_counter        prometheus.Counter
```
[the remark this answers is not in the document]
Comment (carpawell) [reply pkg/metrics/engine.go:11]: done intentionally. i always want deprecated code to scream as loud as possible. is it critical for you? (it gonna like for ~2 weeks i guess before i drop it)

[the remark this answers is not in the document]
Comment (carpawell) [reply pkg/metrics/engine.go:11]: btw, the current master

```bash
▶ golangci-lint run | wc -l
17
```

**Review by cthulhu-rider — APPROVED** [review]

File: pkg/metrics/engine.go (hunk 2 above)
Comment (roman-khimov) [reply pkg/metrics/engine.go:69]: Yeah, I'm not sure it's clear what is meant this way.

File: pkg/metrics/engine.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (roman-khimov) [reply pkg/metrics/engine.go:11]: Worth fixing and I'd not add new ones for sure. We actually have a bit wrong linter setup, it should always complain, not just when something bad is being added.

[the remark this answers is not in the document]
Comment (carpawell) [reply pkg/metrics/engine.go]: fixed (but not agree fully)

File: pkg/metrics/engine.go (hunk 2 above)
Comment (carpawell) [reply pkg/metrics/engine.go]: fixed (single quote)

File: pkg/metrics/engine.go [hunk 4]
```diff
@@ -8,103 +8,197 @@ import (
...
+			Subsystem: engineSubsystem,
+			Name:      "list_containers_time",
+			Help:      "Engine 'list' containers handling time",
+		})
+
+		estimateContainerSizeDuration = prometheus.NewHistogram(prometheus.HistogramOpts{
+			Namespace: storageNodeNameSpace,
+			Subsystem: engineSubsystem,
+			Name:      "estimate_container_size_time",
+			Help:      "Engine 'container' size handling time",
```
Comment (roman-khimov) [inline pkg/metrics/engine.go:55]: Looks weird.

Comment (carpawell) [reply pkg/metrics/engine.go]: fixed, checked one more time the others

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**